lib/packet.ml

```diff
@@ -1489,12 +1488,12 @@ let parse buf =
   (* eprintf "RX: %s\n%!" (to_string dns); *)
   dns
 
-let marshal txbuf dns =
+let marshal dns =
+  let txbuf = Cstruct.create 4096 in
```

Comment:
It's this `Cstruct.create` that could potentially cripple performance for a caller, since there is now choice but to allocate a page for every packet. Previously, the caller of the interface would control memory allocation and could reuse pages from a free pool if it was safe.

Why not just make txbuf an optional function that can be passed in, that does the allocation (defaulting to Cstruct.create)?

OCaml doesn't deal particularly well with rapid page allocation via Bigarrays, since there is also a C-level proxy object that (unnecessarily in our use case) reference counts the memory, so I'm reluctant to make ocaml-dns force memory allocation on the caller unless there's a really good reason.